### PR TITLE
Use latest `sphinx` version on RTD.

### PR DESCRIPTION
**Pull request by hvy**
An attempt to render parameter types in the documentation by using a newer version of `sphinx` on Read the Docs. I followed the note in https://docs.readthedocs.io/en/stable/builds.html#understanding-what-s-going-on but not really sure if this'll work.

### CODE REVIEW

**Comment by hvy** [comment]
The RTD settings must then be updated https://docs.readthedocs.io/en/stable/guides/specifying-dependencies.html#specifying-a-requirements-file.

File: docs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+sphinx==3.0.0
```
Comment (sile) [inline docs/requirements.txt:1]: How about allowing v3.0.0 or later?
```suggestion
sphinx>=3.0.0
```

Comment (hvy) [reply docs/requirements.txt:1]: That makes sense, thanks.

**Comment by toshihikoyanase** [comment]
The parameter types are rendered successfully :tada:
image

**Comment by hvy** [comment]
Just leaving a note but we didn't have to update the path in the RTD dashboard (as mentioned here https://github.com/optuna/optuna/pull/1108#issuecomment-611342241). Maybe because the default path is being used.

**Pull request merged**